### PR TITLE
Switch SD2cadrLib to fastio

### PR DESCRIPTION
**Pull request by AnHardt**
Clean out SD2PinMap.h to the bare bare definitions of the hardware SPI pins.
The definition can easily be overwritten from the boars pinfiles.

### CODE REVIEW

**Comment by thinkyhead** [comment]
I see `Temperature.cpp` wants to `#include "Sd2PinMap.h"` for some reason.

**Comment by AnHardt** [comment]
Yea - surprised me too. But for sure it has to do with the MAX6675 and co.

**Comment by AnHardt** [comment]
Tried to solve that by assuming - a second, unconditional init of the SPI-hardware can't hurt. The pin definitions come from fastio.h now anyway.
And here for the MAX we have the next halve SPI  implementation.

**Comment by thinkyhead** [comment]
Wow. If this is all the change needed, and we have confidence, then I would enjoy merging it and saving more CPU cycles.

**Comment by thinkyhead** [comment]
If the `SDSS` pin is always using "Teensyduino" pin mapping, I suppose we could put a comment on it in all the pins files so we don't forget. And generally make sure all the pin mappings (if out of the ordinary) are commented to be more obvious.

**Comment by AnHardt** [comment]
Let's again have a look on what we remove and how we replace it.

File: Marlin/Sd2PinMap.h [hunk 1]
```diff
@@ -1,453 +0,0 @@
...
-#include "Marlin.h"
-#include "macros.h"
-
-#if ENABLED(SDSUPPORT)
-
-#ifndef Sd2PinMap_h
-#define Sd2PinMap_h
-#include <avr/io.h>
-//------------------------------------------------------------------------------
-/** struct for mapping digital pins */
```
Comment (AnHardt) [inline Marlin/Sd2PinMap.h:39]: All above is uninteresting.

File: Marlin/Sd2PinMap.h [hunk 2]
```diff
@@ -1,453 +0,0 @@
...
-struct pin_map_t {
-  volatile uint8_t* ddr;
-  volatile uint8_t* pin;
-  volatile uint8_t* port;
-  uint8_t bit;
-};
-//------------------------------------------------------------------------------
```
Comment (AnHardt) [inline Marlin/Sd2PinMap.h:46]: The struct used in the functions (at the end of this file - and only there) we want to remove.

File: Marlin/Sd2PinMap.h [hunk 3]
```diff
@@ -1,453 +0,0 @@
...
-
-#undef MOSI_PIN
-#undef MISO_PIN
-#undef SCK_PIN
-// SPI port
-uint8_t const SS_PIN = 53;    // B0
-uint8_t const MOSI_PIN = 51;  // B2
-uint8_t const MISO_PIN = 50;  // B3
-uint8_t const SCK_PIN = 52;   // B1
-
```
Comment (AnHardt) [inline Marlin/Sd2PinMap.h:61]: Hardware dependent pin-number definitions we shift/take to/from fastio.h/pins.h

File: Marlin/Sd2PinMap.h [hunk 4]
```diff
@@ -1,453 +0,0 @@
...
-  {&DDRK, &PINK, &PORTK, 0},  // K0 62
-  {&DDRK, &PINK, &PORTK, 1},  // K1 63
-  {&DDRK, &PINK, &PORTK, 2},  // K2 64
-  {&DDRK, &PINK, &PORTK, 3},  // K3 65
-  {&DDRK, &PINK, &PORTK, 4},  // K4 66
-  {&DDRK, &PINK, &PORTK, 5},  // K5 67
-  {&DDRK, &PINK, &PORTK, 6},  // K6 68
-  {&DDRK, &PINK, &PORTK, 7}   // K7 69
-};
-//------------------------------------------------------------------------------
```
Comment (AnHardt) [inline Marlin/Sd2PinMap.h:134]: Table associating pin-numbers with DataDirectionRegisters, PositionsINregisters, PORts. We don't need that anymore.

File: Marlin/Sd2PinMap.h [hunk 5]
```diff
@@ -1,453 +0,0 @@
...
-#elif defined(__AVR_ATmega644P__)\
-|| defined(__AVR_ATmega644__)\
-|| defined(__AVR_ATmega1284P__)
-// Sanguino
-
```
Comment (AnHardt) [inline Marlin/Sd2PinMap.h:139]: Repetitions for other processor/bord types. -> fastio.h.

File: Marlin/Sd2PinMap.h [hunk 6]
```diff
@@ -1,453 +0,0 @@
...
-  {&DDRC, &PINC, &PORTC, 2}, // C2 46
-  {&DDRD, &PIND, &PORTD, 4}, // D4 47
-  {&DDRG, &PING, &PORTG, 2}, // G2 48
-  {&DDRG, &PING, &PORTG, 1}, // G1 49
-  {&DDRG, &PING, &PORTG, 0}, // G0 50
-};
-#else  // defined(__AVR_ATmega1280__)
-#error "unknown chip"
-#endif  // defined(__AVR_ATmega1280__)
-//------------------------------------------------------------------------------
```
Comment (AnHardt) [inline Marlin/Sd2PinMap.h:398]: More repetitions above.

File: Marlin/Sd2PinMap.h [hunk 7]
```diff
@@ -1,453 +0,0 @@
...
-    else {
-      CBI(*digitalPinMap[pin].port, digitalPinMap[pin].bit);
-    }
-  }
-  else {
-    badPinNumber();
-  }
-}
-#endif  // Sd2PinMap_h
-
```
Comment (AnHardt) [inline Marlin/Sd2PinMap.h:451]: Unused functions. Replaced by the fastio versions.

So - should be save to remove.

File: Marlin/Sd2PinMap.h [hunk 8]
```diff
@@ -1,453 +0,0 @@
...
-  volatile uint8_t* port;
-  uint8_t bit;
-};
-//------------------------------------------------------------------------------
-#if defined(__AVR_ATmega1280__) || defined(__AVR_ATmega2560__) // Mega
-
-// Two Wire (aka I2C) ports
-uint8_t const SDA_PIN = 20;  // D1
-uint8_t const SCL_PIN = 21;  // D0
-
```
Comment (AnHardt) [inline Marlin/Sd2PinMap.h:52]: Will have an other look on the I2C pins: SDA_PIN and SCL_PIN.
...
Both are unused in the rest of the Marlin code.

**Pull request merged**